rviz_common/src/rviz_common/display_factory.cpp

```diff
@@ -128,8 +128,10 @@ QSet<QString> DisplayFactory::getMessageTypes(const QString & class_id)
     while (library != NULL) {
       TiXmlElement * class_element = library->FirstChildElement("class");
       while (class_element) {
-        std::string derived_class = class_element->Attribute("type");
-
+        std::string derived_class;
+        if (class_element->Attribute("type") != NULL) {
```

Comment:
Again, bare or compare to `nullptr`.

Reply:
Go through all the code of RVIZ package, there are many places use "!= NULL" or "==NULL", is it necessary to apply all follow "bare or compare to nullptr"? If yes, I would submit a pr to replace all.

Reply:
You're welcome to do that, but note that you shouldn't touch code in the `rviz` sub folder. We typically fix this kind of style thing as we port to one of `rviz_rendering`, `rviz_common`, `rviz_default_plugins`, or `rviz2` and leave things in `rviz` as-is until we move it.

Reply:
Also, we typically prefer just `if (ptr)`, but if you do use `if (ptr == nullptr)`, according to our style guide we suggest (don't require) `if (nullptr == ptr)` to avoid the typo-bug `if (ptr = nullptr)`. See: https://github.com/ros2/ros2/wiki/Developer-Guide#c